lib/modules/apostrophe-areas/public/js/user.js

```diff
@@ -323,6 +323,41 @@ apos.define('apostrophe-areas', {
       return editors;
     };
 
+    // Loads the virtual areas inside a modal.
+    self.loadVirtualArea = function () {
+      var $virtualAreas = $('[data-virtual-area]');
+      var virtualAreaLength = $virtualAreas.length;
+      var virtualAreaLoadedCount = 0;
+      // Do nothing if there are no virtual areas
+      if ($virtualAreas.length > 0) {
+        // Disable the modal controls till every area is loaded. Clicking on Save fails without
+        // loading the virtual area.
+        apos.modalSupport.toggleModalControls(true);
+        $virtualAreas.each(function (index, element) {
+          var $fieldset = $(element);
+          var areaAttribute = $fieldset.attr('data-virtual-area');
+          if (!areaAttribute) return;
+          var area = JSON.parse(areaAttribute);
+          // Remove the data-attribute after extracting data.
+          $fieldset.removeAttr('data-virtual-area');
+          // Call the Virtual Area Route.
```

Comment:
I believe that's apos.areas.action, you don't have to parse it out. A little surprised it's there in JSON, I wouldn't depend on it.